sdk/python/kfp/compiler/compiler.py

```diff
@@ -394,10 +472,22 @@ def _group_to_template(self, group, inputs, outputs, dependencies):
             })
           else:
             # The value comes from its parent.
-            arguments.append({
-              'name': param_name,
-              'value': '{{inputs.parameters.%s}}' % param_name
-            })
+            # Special handling for recursive subgroup: argument name comes from the existing opsgroup
+            if is_recursive_subgroup:
+              for index, input in enumerate(sub_group.inputs):
+                if param_name == input.name:
+                  break
+              referenced_input = sub_group.recursive_ref.inputs[index-1]
```

Comment:
Is it correct? What happens if index is 0?

Reply:
My mistake. I thought python enumerate starts the index from 1
Fixed.